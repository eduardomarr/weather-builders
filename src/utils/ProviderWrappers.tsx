import React from 'react';
import { ThemeProvider } from 'styled-components/native';
import { WeatherContext } from '../context/weather';
import theme from '../styles/theme';

type WrapperProps = {
  children: React.ReactNode;
};

const mockedContextValues = {
  weather: {
    description: 'description',
    temp: 10,
    feelsLike: 16,
    tempMin: 10,
    tempMax: 16,
    pressure: 10,
    humidity: 50,
    windSpeed: 50,
    district: 'SÃ£o Paulo',
    visibility: 'visibility',
    sunset: 'sunset',
    sunrise: 'sunrise',
    icon: 'icon',
  },

  getCurrentWeather: jest.fn(),
  loading: false,
  setLoading: jest.fn(),
};

export const ThemeWrapper: React.FC<WrapperProps> = ({ children }) => {
  return (
      <ThemeProvider theme={theme}>
        {children}
      </ThemeProvider>
  );
};

export const ContextWrapper: React.FC<WrapperProps> = ({ children }) => (
  <WeatherContext.Provider value={mockedContextValues}>
    {children}
  </WeatherContext.Provider>
);
