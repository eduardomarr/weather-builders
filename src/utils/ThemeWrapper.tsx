import React from 'react';
import { ThemeProvider } from 'styled-components/native';
import theme from '../styles/theme';

type WrapperProps = {
  children: React.ReactNode;
};

export const ThemeWrapper: React.FC<WrapperProps> = ({ children }) => {
  return (
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
  );
};
