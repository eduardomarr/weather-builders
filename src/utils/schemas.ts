import * as Yup from 'yup';

export const signInSchema = Yup.object().shape({
  email: Yup.string().required('E-mail é obrigatório').email('E-mail Inválido!'),
  password: Yup.string().required('Senha é obrigatório'),
});
