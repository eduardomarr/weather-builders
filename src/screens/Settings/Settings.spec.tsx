import React from 'react';
import { ThemeWrapper } from '../../utils/ProviderWrappers';

import { Settings } from '.';

import { render } from '@testing-library/react-native';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';


jest.mock('@react-navigation/native', () => {
  return {
    useNavigation: () => {
      return {
        navigate: jest.fn(),
      };
    },
  };
});

describe('Settings Screen', () => {

  it('should be render correctly', () => {

    const { getByTestId } = render(
      <Settings />,
      {wrapper: ThemeWrapper}
    );

    expect(getByTestId('container')).toBeOnTheScreen();

  });

  it('should be render Screen List', () => {

    const { getByTestId } = render(
      <Settings />,
      {wrapper: ThemeWrapper}
    );

    expect(getByTestId('screen-list')).toBeOnTheScreen();

  });

  it('should be styled correctly', () => {
    const tree = renderer.create(
    <ThemeWrapper>
      <Settings />
    </ThemeWrapper>
    ).toJSON() as ReactTestRendererJSON;

      expect(tree).toMatchSnapshot();

  });

});
