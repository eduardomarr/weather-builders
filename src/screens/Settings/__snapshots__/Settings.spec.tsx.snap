// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Screen should be styled correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#090932",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "space-between",
        "paddingTop": 80,
      },
    ]
  }
  testID="container"
>
  <View
    style={
      Array [
        Object {
          "width": "100%",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginBottom": 8,
          "opacity": 1,
          "paddingBottom": 0,
          "paddingLeft": 24,
          "paddingRight": 24,
          "paddingTop": 0,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingBottom": 8,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 8,
            },
          ]
        }
        testID="screen-list"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Manrope-Medium",
                "fontSize": 35,
              },
            ]
          }
        >
          Sobre
        </Text>
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="28"
          bbWidth="28"
          fill="#969CB2"
          focusable={false}
          height="28"
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Array [
                Object {},
              ],
              Object {
                "flex": 0,
                "height": 28,
                "width": 28,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
          width="28"
          xmlns="http://www.w3.org/2000/svg"
        >
          <RNSVGGroup
            fill={4288060594}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M0 0h24v24H0z"
              fill={null}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M13.172 12l-4.95-4.95 1.414-1.414L16 12l-6.364 6.364-1.414-1.414z"
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
</View>
`;
